utilities/infra.py

```diff
@@ -116,7 +108,9 @@ def s3_endpoint_secret(
         name=name,
         namespace=namespace,
         annotations={"opendatahub.io/connection-type": "s3"},
-        data_dict=get_s3_secret_dict(
+        # the labels are needed to set the secret as data connection by odh-model-controller
+        label={"opendatahub.io/managed": "true", "opendatahub.io/dashboard": "true"},
```

Comment:
I don't really like the fact that we're putting a label here that signals this resource was created by the dashboard when it isn't so. If we want to test in a way that is decouple from the dashboard we shouldn't rely on its logic and behaviour IMHO

Reply:
i do not like it either, see https://github.com/opendatahub-io/odh-model-controller/issues/348
but this how the product is working now